### ISSUE TITLE
Accessibility issue with autocomplete component when there are child elements in the list.

### ISSUE THREAD

**Issue by knamburigsa** [GSA/sam-design-system#1451]
**Components Involved**
Please list the common component(s) involved that you're experiencing an issue with
Autocomplete-Search component. 
https://github.com/GSA/sam-design-system/tree/master/libs/packages/components/src/lib/autocomplete-search

**Expected Behavior**
How you expect the listed component(s) to function
When we have nested children for the list of options jaws should be able to read them correctly when using arrows keys for navigation.

**Actual Behavior**
How the listed component(s) actually function
Jaws is not able to read the list of options correctly when using arrow keys for navigation.

**Your Environment**
Angular Version: X.Y.Z
@gsa-sam packages version: X.Y.Z (Please list individually if all packages are not same version)

**Ticket Number**
Please provide your JIRA ticket number that is dependent on this fix. If there is none, please create
one to validate the issue once the fix has been released. DO NOT provide the url, please ONLY
provide the ticket number.
IAEMOD-13127

**Additional context**
Screenshots / recordings, minimal reproduction on stackblitz, steps to reproduce, code snippets, etc.
you can reproduce the issue on sam.gov search page.

https://github.com/GSA/sam-design-system/assets/92172714/9db7904d-6570-4ffd-9378-3932844cade6

**Comment by knamburigsa** [comment]
Not able to reproduce this anymore.

**Comment by yerramshilpa** [comment]
The issue is resolved in @gsa-sam/components@17.0.10
